projects/js-packages/components/components/text/stories/Text-MDX-Documentation.mdx

```diff
@@ -0,0 +1,171 @@
+import dedent from 'ts-dedent';
+import Text, { SPACING_VALUES, BOX_MODEL_VALUES, H2, H3, Title } from '../index.jsx';
+
+<Meta title="JS Packages/Components/Text/Readme" />
+
+# Text
+Set of React jetpack-components intended to facilitate when dealing with rendering text content.
+
+## <Text /\>
+
+`Text` is the primary component that renders text, supporting the following features:
+
+* **Typography**: provided by the ThemeProvider component.
+* **Spacing**: A simple range of shorthand classes to modify the text element's margin and padding.
+
+### Props
+
+#### variant
+
+**Type**: `headline-medium`, `headline-small`, `headline-small-regular`, `title-medium`, `title-small`, `body`, `body-small`, `body-extra-small`, `label`.
+
+Define the text Typography variant among the options offered by the ThemeProvider component.
+
+<Source
+  language='jsx'
+  code={dedent`
+    import Text from '@automattic/jetpack-components';
+    
+    function Hello() {
+      return (
+        <Text variant="headline-medium">
+          Hello, Text component!
+        </Text>
+      );
+    }
+  `}
+/>
+
+#### Spacing properties
+
+**Type**: number, one of `0`, `1`, `2`, `3`, `4`, `5`, `6`, `7`, `8`.
+
+Use the following properties to define margin and padding of the text. The unit is defined by the `--spacing-base` value.
+You can see how it works playing with the [Box Model](./?path=/story/js-packages-components-text--box-model) story.
+
+Prop | Description
+---|---
+**m** | margin
+**mt** | margin-top
+**mr** | margin-right
+**mb** | margin-bottom
+**ml** | margin-left
+**mx** | margin left and right
+**my** | margin top and bottom
+**p** | padding
+**pt** | padding-top
+**pr** | padding-right
+**pb** | padding-bottom
+**pl** | padding-left
+**px** | padding left and right
+**py** | padding top and bottom
+
+<Source
+  language='jsx'
+  code={dedent`
+    import Text from '@automattic/jetpack-components';
+    
+    function HelloBoxModel() {
+      return (
+        <Text mt={ 2 } px={ 4 }>
+          Hello, Text component!
+        </Text>
+      );
+    }
+  `}
+/>
+
+#### component
+
+**Type**: `elementType`, `string`.
+
+Force an specific tag (`<span />`, `<div />`) or use a custom component that will receive className and children.
+
+<Source
+  language='jsx'
+  code={dedent`
+    import Text from '@automattic/jetpack-components';
+    
+    function CustomComponent( { children, className } ) {
+      // className, provided by Text, is "custom-classname"
+      return (
+        <div className={ className }>{ children }</div>
+      );
+    }
+    
+    // Use a custom component when rendering the <Text />,
+    // it passes down the className property to it.
+    function CustomTextComponent() {
+      return (
+        <Text
+          className="custom-classname"
+          component={ CustomComponent }
+        >
+          Hello, Custom Text component!
+        </Text>
+      );
+    }
+  `}
+/>
```

Comment:
Can we add an code example about the tag usage, ex. `component="span"` ?

Reply:
<img width="643" alt="image" src="https://user-images.githubusercontent.com/77539/161161510-925c5bea-c2fc-4279-9bcc-5bde5dac507c.png">

https://github.com/Automattic/jetpack/pull/23725/commits/7833af3bb2bc7de87d1357f4cc649b8c1fe6ad95